Objects/abstract.c

```diff
@@ -821,7 +821,7 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
         Py_DECREF(result);
 
         if (op_slot == NB_SLOT(nb_rshift) && \
-                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
+                PyCFunction_Check(v) && strcmp(((PyCFunctionObject *)v) -> m_ml -> ml_name, "print") == 0) {
```

Comment:
When working with user supplied strings, we generally prefer `strncmp` to `strcmp`, which would give:

    strncmp(((PyCFunctionObject *)v) -> m_ml -> ml_name, "print", 6)

for the comparison subexpression.